api/src/main/java/ca/bc/gov/educ/api/gradbusiness/service/v2/SchoolDetailsService.java

```diff
@@ -0,0 +1,120 @@
+
+	@Autowired
+	public SchoolDetailsService(EducGradBusinessApiConstants constants, WebClient webClient, RESTService restService, JsonTransformer jsonTransformer, EducGraduationApiConstants educGraduationApiConstants) {
+        this.constants = constants;
+        this.webClient = webClient;
+        this.restService = restService;
+        this.jsonTransformer = jsonTransformer;
+		this.educGraduationApiConstants = educGraduationApiConstants;
+	}
+
+	public ResponseEntity<byte[]> getSchoolReportPDFByMincode(String mincode, String type) {
+		Calendar cal = Calendar.getInstance(TimeZone.getTimeZone("PST"), Locale.CANADA);
+		int year = cal.get(Calendar.YEAR);
+		String month = String.format("%02d", cal.get(Calendar.MONTH) + 1);
+		try {
+			List<School> schoolDetails = getSchoolDetails(mincode);
+			if (schoolDetails.isEmpty()) {
+				return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+			}
+			String schoolOfRecordId = schoolDetails.get(0).getSchoolId();
+
+			var result = restService.get(String.format(educGraduationApiConstants.getSchoolReportBySchoolIdAndReportType(), schoolOfRecordId,type), InputStreamResource.class);
+			byte[] response = new byte[0];
+			if (result != null) {
+				response = result.getInputStream().readAllBytes();
+			}
+
+			return handleBinaryResponse(response, EducGradBusinessUtil.getFileNameSchoolReports(mincode,year,month,type,MediaType.APPLICATION_PDF), MediaType.APPLICATION_PDF);
+		} catch (Exception e) {
+			logger.error("Error getting school report PDF by mincode: {}", e.getMessage());
+			return getInternalServerErrorResponse(e);
+		}
+	}
+
+	/**
+	 * Gets internal server error response.
+	 *
+	 * @param t the t
+	 * @return the internal server error response
+	 */
+	protected ResponseEntity<byte[]> getInternalServerErrorResponse(Throwable t) {
+		ResponseEntity<byte[]> result;
+
+		Throwable tmp = t;
+		String message;
+		if (tmp.getCause() != null) {
+			tmp = tmp.getCause();
+			message = tmp.getMessage();
+		} else {
+			message = tmp.getMessage();
+		}
+		if (message == null) {
+			message = tmp.getClass().getName();
+		}
+
+		result = ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(message.getBytes());
+		return result;
+	}
+
+
+	private ResponseEntity<byte[]> handleBinaryResponse(byte[] resultBinary, String reportFile, MediaType contentType) {
+		ResponseEntity<byte[]> response;
+		if (resultBinary.length > 0) {
+			String fileType = contentType.getSubtype().toUpperCase();
+			logger.debug("Sending {} response {} KB", fileType, resultBinary.length / (1024));
+			HttpHeaders headers = new HttpHeaders();
+			headers.add("Content-Disposition", "inline; filename=" + reportFile);
+			response = ResponseEntity
+					.ok()
+					.headers(headers)
+					.contentType(contentType)
+					.body(resultBinary);
+		} else {
+			response = ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+		}
+		return response;
+	}
+
+	public List<School> getSchoolDetails(String mincode) {
```

Comment:
This can be called from the existing SchoolService class. A v2 version isn't needed